include/flashinfer/pos_enc.cuh

```diff
@@ -410,6 +411,21 @@ cudaError_t BatchQKApplyRotary(DType* q, DType* k, DType* q_rope, DType* k_rope,
   return cudaSuccess;
 }
 
+template <typename DType, typename IdType>
+cudaError_t BatchQKApplyRotaryInPlace(DType* __restrict__ q, DType* __restrict__ k,
```

Comment:
btw, we don't really need this, `__restrict__` only matters as the __global__ function arguments, and it won't make effect here.

Reply:
Hi @yzh119 yes, it's no use here since `BatchQKApplyRotaryInPlace` is only used by `_FlashInferBatchQKApplyRotaryInPlace`. I thought if `BatchQKApplyRotaryInPlace` is a public interface exposed by flashinfer, the `__restrict__` modifier also serves as a hint to human that `q` and `k` should be distinct ptrs. You can remove the `__restrict__` modifier if you think the declaration is ok.